sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -191,47 +191,51 @@ public void updateAddresses(final URI serverKey) {
         Utils.checkNotNullOrThrow(request, "request", "");
         Utils.checkNotNullOrThrow(partitionKeyRangeIdentity, "partitionKeyRangeIdentity", "");
 
-        if (forceRefreshPartitionAddresses) {
-            // forceRefreshPartitionAddresses==true indicates we are requesting the latest
-            // Replica addresses from the Gateway
-            // There are a couple of cases (for example when getting 410/0 after a split happened for the parent
-            // partition when just refreshing addresses isn't sufficient (because the Gateway in its cache)
-            // might also not know about the partition split that happened
-            // to recover from this condition the client would need to either trigger a PKRange cache refresh
-            // on the client or force the Gateway CollectionRoutingMap to be refreshed (so that the Gateway gets
-            // aware of the split and latest EPK map.
-            // Due to the fact that forcing the CollectionRoutingMap to be refreshed in Gateway there is additional
-            // load on the ServiceFabric naming service we want to throttle how often we would force the collection
-            // routing map refresh
-            // These are the throttle conditions: We will only enforce collection routing map to be refreshed
-            // - if there has been at least 1 attempt to just refresh replica addresses without forcing collection
-            //   routing map refresh on the physical partition before
-            // - only one request per Container to force collection routing map refresh is allowed every 30 seconds
-            //
-            // The throttling logic is implemented in  `ForcedRefreshMetadata.shouldIncludeCollectionRoutingMapRefresh`
-            ForcedRefreshMetadata forcedRefreshMetadata = this.lastForcedRefreshMap.computeIfAbsent(
-                partitionKeyRangeIdentity.getCollectionRid(),
-                (colRid) -> new ForcedRefreshMetadata());
-
-            if (request.forceCollectionRoutingMapRefresh) {
-                forcedRefreshMetadata.signalCollectionRoutingMapRefresh(
-                    partitionKeyRangeIdentity,
-                    true);
-            } else if (forcedRefreshMetadata.shouldIncludeCollectionRoutingMapRefresh(partitionKeyRangeIdentity)) {
-                request.forceCollectionRoutingMapRefresh = true;
+        // For resolving master resource address, collectionRid is null.
+        // In those cases, we cannot do this optimization, hence the check.
+        if (partitionKeyRangeIdentity.getCollectionRid() != null) {
```

Comment:
Should we move this new logic in server partition method instead of this generic method, logically it is only for server partition , thoughts ?

Reply:
I like Naveen's proposal - have included the change along those lines in this PR - https://github.com/Azure/azure-sdk-for-java/pull/22950 - please take a look

Reply:
@FabianMeiswinkel  @simplynaveen20  - I added one comment on the above PR mentioned by Fabian. 
Also, since this is a bug open in master code right now, we need to fix it before the next release. 
If we are sure that the above PR https://github.com/Azure/azure-sdk-for-java/pull/22950 will get merged before next release, then we should be good, otherwise we may have to merge this hotfix separately.